src/mlpack/methods/ann/layer/channel_shuffle_impl.hpp

```diff
@@ -0,0 +1,139 @@
+/**
+ * @file methods/ann/layer/channe_shuffle_impl.hpp
+ * @author Abhinav Anand
+ *
+ * Implementation of the channel shuffle function as an individual layer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license. You should have received a copy of the
+ * 3-clause BSD license along with mlpack. If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_CHANNEL_SHUFFLE_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_CHANNEL_SHUFFLE_IMPL_HPP
+
+// In case it hasn't yet been included.
+#include "channel_shuffle.hpp"
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+
+template<typename InputDataType, typename OutputDataType>
+ChannelShuffle<InputDataType, OutputDataType>::
+ChannelShuffle():
+    inRowSize(0),
+    inColSize(0),
+    depth(0),
+    groupCount(0),
+    batchSize(0)
+{
+  // Nothing to do here.
+}
+
+template<typename InputDataType, typename OutputDataType>
+ChannelShuffle<InputDataType, OutputDataType>::
+ChannelShuffle(
+    const size_t inRowSize,
+    const size_t inColSize,
+    const size_t depth,
+    const size_t groupCount):
+    inRowSize(inRowSize),
+    inColSize(inColSize),
+    depth(depth),
+    groupCount(groupCount),
+    batchSize(0)
+{
+  if (depth % groupCount != 0)
+  {
+    Log::Fatal << "Number of channels must be divisible by groupCount.!" << std::endl;
+  }
+}
+
+template<typename InputDataType, typename OutputDataType>
+template<typename eT>
+void ChannelShuffle<InputDataType, OutputDataType>::Forward(
+    const arma::Mat<eT>& input, arma::Mat<eT>& output)
+{
+  batchSize = input.n_cols;
+
+  if (output.is_empty())
+    output.set_size(inRowSize * inColSize * depth, batchSize);
+  else
+  {
+    assert(output.n_rows == inRowSize * inColSize * depth);
+    assert(output.n_cols == batchSize);
+  }
+
+
+  arma::cube inputAsCube(const_cast<arma::Mat<eT>&>(input).memptr(),
+      inRowSize, inColSize, depth * batchSize, false, false);
+  arma::cube outputAsCube(output.memptr(), inRowSize, inColSize,
+                          depth * batchSize, false, true);
```

Comment:
```suggestion
  arma::cube outputAsCube(output.memptr(), inRowSize, inColSize,
      depth * batchSize, false, true);
```

Just to use the same style as the line right above.